lib/crawler.js

```diff
@@ -50,124 +40,276 @@ var Crawler = function(initialURL) {
-    // One request will be spooled per tick, up to the concurrency threshold.
-    crawler.interval        = 250;
-
-    // Maximum request concurrency. Be sensible. Five ties in with node's
-    // default maxSockets value.
-    crawler.maxConcurrency  = 5;
-
-    // Maximum time we'll wait for headers
-    crawler.timeout         = 5 * 60 * 1000;
-
-    // Maximum time we'll wait for async listeners.
-    crawler.listenerTTL     = 10 * 1000;
-
-    // User Agent
-    crawler.userAgent =
-            "Node/" + MetaInfo.name + " " + MetaInfo.version +
-            " (" + MetaInfo.repository.url + ")";
-
-    // Queue for requests - FetchQueue gives us stats and other sugar
-    // (but it's basically just an array)
-    crawler.queue           = new FetchQueue();
-
-    // Do we exclude certain URL's based on rules found in robots.txt?
-    crawler.respectRobotsTxt = true;
-
-    // Should we update crawler.host if the first response is a redirect to another domain.
-    crawler.allowInitialDomainChange = false;
-
-    // Set Accept-Encoding header and automatically decompress HTTP responses
-    // based on Content-Encoding header
-    crawler.decompressResponses = true;
-
-    // Decode HTTP responses based on their Content-Type header or any
-    // inline charset definition
-    crawler.decodeResponses = false;
-
-    // Do we filter by domain?
-    // Unless you want to be crawling the entire internet, I would
-    // recommend leaving this on!
-    crawler.filterByDomain  = true;
-
-    // Do we scan subdomains?
-    crawler.scanSubdomains  = false;
-
-    // Treat WWW subdomain the same as the main domain (and don't count
-    // it as a separate subdomain)
-    crawler.ignoreWWWDomain = true;
-
-    // Or go even further and strip WWW subdomain from domains altogether!
-    crawler.stripWWWDomain  = false;
-
-    // Internal cachestore
-    crawler.cache           = null;
-
-    // Use an HTTP Proxy?
-    crawler.useProxy        = false;
-    crawler.proxyHostname   = "127.0.0.1";
-    crawler.proxyPort       = 8123;
-    crawler.proxyUser       = null;
-    crawler.proxyPass       = null;
-
-    // Support for HTTP basic auth
-    crawler.needsAuth       = false;
-    crawler.authUser        = "";
-    crawler.authPass        = "";
-
-    // Support for retaining cookies for parse duration
-    crawler.acceptCookies   = true;
-    crawler.cookies         = new CookieJar();
-
-    // Support for custom headers...
-    crawler.customHeaders   = {};
-
-    // Domain Whitelist
-    // We allow domains to be whitelisted, so cross-domain requests can be made.
-    crawler.domainWhitelist = [];
-
-    // Keep track of what hosts (hostname and port - if port is non-standard)
-    // we've fetched from
-    crawler.touchedHosts = [];
-
-    // Supported Protocols
-    crawler.allowedProtocols = [
+    /**
+     * Controls which URL to request first
+     * @type {String}
+     */
+    this.initialURL = initialURL;
```

Comment:
Assigning properties directly to `this` rather than to a reference to `this` (ie. `crawler`) makes it so that JSDoc can understand what's a class member